File: client/src/app/models/birthday.ts
export interface Birthday {
  id: string;
  fullName: string;
  date: Date;
  note?: string;
  age: number;
  daysUntilBirthday: number;
}

export interface CreateBirthdayDto {
  fullName: string;
  date: Date;
  note?: string | null;
}

export interface EditBirthdayDto {
  id: string;
  fullName: string;
  date: Date;
  note?: string | null;
}

File: backend/src/Domain/Birthdays/Birthday.cs
using Domain.Common;
using Domain.Users;

namespace Domain.Birthdays;

public sealed class Birthday : Entity
{
    public Guid UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public string? Note { get; set; }

    public User? User { get; set; }
}
